### DISCUSSION TITLE
Gen Signal & Gen Force (address 242 & 326) / Help to write it correctly

### DISCUSSION THREAD

**Discussion by man55** [kellerza/sunsynk#328] [Ideas]
I found the necessary addresses for control it.
Address 242 is used to configure the `grid signal` and `gen signal` options. I checked it, it works.
```
SIGNAL_OPTIONS = {
    0: "No Grid or Gen",
    1: "Allow Grid",
    2: "Allow Gen",
    3: "Allow Grid & Gen",
}
SENSORS += (
  SelectRWSensor(242, "GRID&GEN Signal", options=SIGNAL_OPTIONS, bitmask=0x03),
)
```
But it is much more logical and in accordance with the settings in the inverter itself that we need to have not a `SelectRWSensor`, but two `SwitchRWSensor`.
And in my case I really only need one sensor to turn the generator signal on and off.

Now this part of the configuration looks like this for me:
```
SIGNAL_OPTIONS = {
    1: "Disable",
    3: "Enable",
}
....
SelectRWSensor(242, "GEN Signal", options=SIGNAL_OPTIONS, bitmask=0x03),
....
```
**How to properly turn this into a switch (`SwitchRWSensor`)?**

-----
And another great option is the ability to force the generator to start. Now for this I need to go directly to the inverter.
But we have an address `326` to control this future. 
The only problem is that we need to reach bit 13.
That is, we need a SwitchRWSensor to reach bit 13 at address 326. 
But how to do it correctly?

Here's what it looks like in another integration:
```
  - platform: modbus_controller            # 326 Toggle Force Generator
    use_write_multiple: true
    modbus_controller_id: inverter
    name: "${friendly_name} Toggle Force Generator"
    id: inverter_esphome_toggle_force_generator
    register_type: holding
    address: 326
    bitmask: 8192 # 2^13 to target bit 13
    icon: "mdi:toggle-switch"
```
![Снимок экрана 2024-08-26 171547](https://github.com/user-attachments/assets/ae886a8d-ad0e-4f69-992c-32e9cf9b07d4)

**Comment by kellerza** [comment]
In Python you can use `pow(2,13)` or left-shift.

```python
SwitchRWSensor(326, "Force Gen", on=1<<13, bitmask=1<<13)
```
